crates/matrix-sdk-crypto/src/store/memorystore.rs

```diff
@@ -138,6 +138,11 @@ impl CryptoStore for MemoryStore {
         Ok(self.next_batch_token.read().await.clone())
     }
 
+    async fn save_pending_changes(&self, _changes: PendingChanges) -> Result<()> {
+        // TODO(bnjbvr) why didn't save_changes save the account?
```

Comment:
The account is, or perhaps was, only loaded when you create the `OlmMachine`. Once that's done the `Account` lives and dies with the `OlmMachine`.

This made it unnecessary for the `MemoryStore` to persist the account.

Reply:
Interesting. With this new design, the `ReadOnlyAccount` would only live transiently:

- either read from the database, available from the `cache()`
- or we plan to mutably use it, which implies the use of a `transaction()`

In that case, we would need to save it for the memory store.

(Note that, at the end of the day, the memory store *could* be exactly the same thing as the in-memory cache :zany_face: and since a cross-process lock for the memory store doesn't make sense, all the methods in the memory stub could do strictly nothing, since the cache would keep the values in memory :brain:)